### PR TITLE
Timer2

### PR DESCRIPTION
**Pull request by AashnaNarang**
- Renamed getId to getElevatorId
- changed floorsubsystem + elevatorsubsystem to use configurations file instead of constructor for numFloors
- Added a class that can be used as a timer 
     - can cancel timer, start it,

### CODE REVIEW

File: src/events/StationaryEvent.java [hunk 1]
```diff
@@ -3,27 +3,22 @@
...
+	private int currentFloor;
 
 	/**
 	 * 
 	 * @param floorPort The port floor events should be sent to.
 	 * @param id The id of the elevator
 	 */
 	
-	public StationaryEvent(int floorPort, int id) {
+	public StationaryEvent(int floorPort, int id, int currentFloor) {
```
Comment (krishane2018) [inline src/events/StationaryEvent.java:19]: Add javadoc for new param

File: src/main/Scheduler.java [hunk 2]
```diff
@@ -64,13 +74,52 @@ public Scheduler(int floorEventPort, int arrPort, int destPort, int floorPort, i
...
+		ArrayList<Event> destEventRemove = new ArrayList<Event>();
+
+		for(FloorEvent e: sentFloorEvents) {
+			if(e.getElevatorId() == t.getElevatorId()) {
+				e.setErrorCode(0);
+				floorEventRemove.add(e);
+			}
+		}
+		floorEvents.addAll(floorEventRemove);
+		sentFloorEvents.removeAll(floorEventRemove);
```
Comment (krishane2018) [inline src/main/Scheduler.java:110]: Can you remove events in the for loop instead of doing this?

File: src/timers/SchedulerTimer.java [hunk 3]
```diff
@@ -0,0 +1,57 @@
...
+public class SchedulerTimer extends NetworkCommunicator implements Runnable {
+	private Scheduler scheduler;
+	private int numFloors;
+	private boolean beforeArrivedAtSrcFloor;
+	private int elevatorId;
+	private int timerPort;
+	private DatagramSocket sendTimerSocket;
+	private Thread t;
+	
+	public SchedulerTimer(Scheduler scheduler, String name, boolean beforeArrivedAtSrcFloor, int elevatorId, int timerPort) {
```
Comment (krishane2018) [inline src/timers/SchedulerTimer.java:20]: Where do we use name?

File: src/main/Scheduler.java (hunk 2 above)
Comment (AashnaNarang) [reply src/main/Scheduler.java:110]: nope, sadly it causes a really ugly error

File: src/timers/SchedulerTimer.java (hunk 3 above)
Comment (AashnaNarang) [reply src/timers/SchedulerTimer.java:20]: to pass in to the super's constructor

**Pull request closed without merging**